### ISSUE TITLE
安装5.2时验证不通过

### ISSUE THREAD

**Issue by flben233** [v2fly/fhs-install-v2ray#261]
使用指令及报错信息如下，系统为Debian10，
已经安装了4.45.0，想使用此脚本更新到最新的5.2.0
直接运行此脚本安装的版本为5.1.0，此时能够正常安装

```
root@WoozyMinty-VM:~# ./install-release.sh --version v5.2.0
info: Installing V2Ray v5.2.0 for x86_64
Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.0/v2ray-linux-64.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 11.0M  100 11.0M    0     0  2818k      0  0:00:04  0:00:04 --:--:-- 4423k
Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.0/v2ray-linux-64.zip.dgst
error: Check failed! Please check your network or try again.
removed: /tmp/tmp.zsop9W1Tjf
```

**Comment by IceCodeNew** [comment]
看起来是网络问题，指定 5.2.1 再试试呢？

**Comment by flben233** [comment]
> 看起来是网络问题，指定 5.2.1 再试试呢？
错误是一样的
```
root@shirakawatyu:~# ./install-release.sh --version v5.2.1
info: Installing V2Ray v5.2.1 for x86_64
Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 11.0M  100 11.0M    0     0  5711k      0  0:00:01  0:00:01 --:--:-- 7662k
Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
error: Check failed! Please check your network or try again.
removed: /tmp/tmp.oC8MYNKAWd
```

**Comment by IceCodeNew** [comment]
`bash -x https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst`

**Comment by flben233** [comment]
> `bash -x ./install-release.sh --version v5.2.1`
> 
> 试试呢

结果是这样的
```
root@shirakawatyu:~# bash -x ./install-release.sh --version v5.2.1
+ DAT_PATH=/usr/local/share/v2ray
+ JSON_PATH=/usr/local/etc/v2ray
+ main --version v5.2.1
+ check_if_running_as_root
+ [[ 0 -ne 0 ]]
+ identify_the_operating_system_and_architecture
++ uname
+ [[ Linux == \L\i\n\u\x ]]
+ case "$(uname -m)" in
++ uname -m
+ MACHINE=64
+ [[ ! -f /etc/os-release ]]
+ [[ -f /.dockerenv ]]
+ grep -q 'docker\|lxc' /proc/1/cgroup
+ [[ -d /run/systemd/system ]]
+ true
++ type -P apt
+ [[ -n /usr/bin/apt ]]
+ PACKAGE_MANAGEMENT_INSTALL='apt -y --no-install-recommends install'
+ PACKAGE_MANAGEMENT_REMOVE='apt purge'
+ package_provide_tput=ncurses-bin
+ judgment_parameters --version v5.2.1
+ [[ 2 -gt 0 ]]
+ case "$1" in
+ VERSION=v5.2.1
+ break
+ install_software ncurses-bin tput
+ package_name=ncurses-bin
+ file_to_detect=tput
+ type -P tput
+ return
++ tput setaf 1
+ red=''
++ tput setaf 2
+ green=''
++ tput setaf 6
+ aoi=''
++ tput sgr0
+ reset=''
+ [[ '' -eq 1 ]]
+ [[ '' -eq 1 ]]
+ [[ '' -eq 1 ]]
++ mktemp -d
+ TMP_DIRECTORY=/tmp/tmp.2PjHux83CJ
+ ZIP_FILE=/tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
+ [[ '' -eq 1 ]]
+ install_software curl curl
+ package_name=curl
+ file_to_detect=curl
+ type -P curl
+ return
+ get_version
+ [[ -n v5.2.1 ]]
+ RELEASE_VERSION=v5.2.1
+ return 2
+ NUMBER=2
+ [[ 2 -eq 0 ]]
+ [[ '' -eq 1 ]]
+ [[ 2 -eq 2 ]]
++ uname -m
+ echo 'info: Installing V2Ray v5.2.1 for x86_64'
info: Installing V2Ray v5.2.1 for x86_64
+ download_v2ray
+ DOWNLOAD_LINK=https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
+ echo 'Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip'
Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
+ curl -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
++ type -P curl
+ /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 11.0M  100 11.0M    0     0  15.8M      0 --:--:-- --:--:-- --:--:-- 57.1M
+ echo 'Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst'
Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
+ curl -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
++ type -P curl
+ /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
++ cat /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
+ [[ MD5= 73f41a7eacd77fe99fdf33c56ea4b8d0
SHA1= e47119495c4e73658f13e817b0b552ac17f8cf5d
SHA2-256= 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
SHA2-512= 12dcb222b7e683d856f94b59ac1013943a55a413c2087b9ccdff639b420826edaadbd44c236f31a6f307a94adc3c3ae955d602219a6373019309c8012e6d2fd0 == \N\o\t\ \F\o\u\n\d ]]
+ for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
++ sed 's/ .*//'
++ md5sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
+ SUM=73f41a7eacd77fe99fdf33c56ea4b8d0
++ uniq
++ grep 73f41a7eacd77fe99fdf33c56ea4b8d0 -o -a
++ grep MD5 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
+ CHECKSUM=73f41a7eacd77fe99fdf33c56ea4b8d0
+ [[ 73f41a7eacd77fe99fdf33c56ea4b8d0 != \7\3\f\4\1\a\7\e\a\c\d\7\7\f\e\9\9\f\d\f\3\3\c\5\6\e\a\4\b\8\d\0 ]]
+ for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
++ sed 's/ .*//'
++ sha1sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
+ SUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
++ uniq
++ grep e47119495c4e73658f13e817b0b552ac17f8cf5d -o -a
++ grep SHA1 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
+ CHECKSUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
+ [[ e47119495c4e73658f13e817b0b552ac17f8cf5d != \e\4\7\1\1\9\4\9\5\c\4\e\7\3\6\5\8\f\1\3\e\8\1\7\b\0\b\5\5\2\a\c\1\7\f\8\c\f\5\d ]]
+ for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
++ sed 's/ .*//'
++ sha256sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
+ SUM=56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
++ uniq
++ grep 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a -o -a
++ grep SHA256 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
+ CHECKSUM=
+ [[ 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a != '' ]]
+ echo 'error: Check failed! Please check your network or try again.'
error: Check failed! Please check your network or try again.
+ return 1
+ [[ 1 -eq 1 ]]
+ rm -r /tmp/tmp.2PjHux83CJ
+ echo 'removed: /tmp/tmp.2PjHux83CJ'
removed: /tmp/tmp.2PjHux83CJ
+ exit 1
```

**Comment by IceCodeNew** [comment]
> > `bash -x ./install-release.sh --version v5.2.1`
> > 试试呢
> 
> 结果是这样的
> 
> ```
> root@shirakawatyu:~# bash -x ./install-release.sh --version v5.2.1
> + DAT_PATH=/usr/local/share/v2ray
> + JSON_PATH=/usr/local/etc/v2ray
> + main --version v5.2.1
> + check_if_running_as_root
> + [[ 0 -ne 0 ]]
> + identify_the_operating_system_and_architecture
> ++ uname
> + [[ Linux == \L\i\n\u\x ]]
> + case "$(uname -m)" in
> ++ uname -m
> + MACHINE=64
> + [[ ! -f /etc/os-release ]]
> + [[ -f /.dockerenv ]]
> + grep -q 'docker\|lxc' /proc/1/cgroup
> + [[ -d /run/systemd/system ]]
> + true
> ++ type -P apt
> + [[ -n /usr/bin/apt ]]
> + PACKAGE_MANAGEMENT_INSTALL='apt -y --no-install-recommends install'
> + PACKAGE_MANAGEMENT_REMOVE='apt purge'
> + package_provide_tput=ncurses-bin
> + judgment_parameters --version v5.2.1
> + [[ 2 -gt 0 ]]
> + case "$1" in
> + VERSION=v5.2.1
> + break
> + install_software ncurses-bin tput
> + package_name=ncurses-bin
> + file_to_detect=tput
> + type -P tput
> + return
> ++ tput setaf 1
> + red=''
> ++ tput setaf 2
> + green=''
> ++ tput setaf 6
> + aoi=''
> ++ tput sgr0
> + reset=''
> + [[ '' -eq 1 ]]
> + [[ '' -eq 1 ]]
> + [[ '' -eq 1 ]]
> ++ mktemp -d
> + TMP_DIRECTORY=/tmp/tmp.2PjHux83CJ
> + ZIP_FILE=/tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> + [[ '' -eq 1 ]]
> + install_software curl curl
> + package_name=curl
> + file_to_detect=curl
> + type -P curl
> + return
> + get_version
> + [[ -n v5.2.1 ]]
> + RELEASE_VERSION=v5.2.1
> + return 2
> + NUMBER=2
> + [[ 2 -eq 0 ]]
> + [[ '' -eq 1 ]]
> + [[ 2 -eq 2 ]]
> ++ uname -m
> + echo 'info: Installing V2Ray v5.2.1 for x86_64'
> info: Installing V2Ray v5.2.1 for x86_64
> + download_v2ray
> + DOWNLOAD_LINK=https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> + echo 'Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip'
> Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> + curl -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> ++ type -P curl
> + /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 100 11.0M  100 11.0M    0     0  15.8M      0 --:--:-- --:--:-- --:--:-- 57.1M
> + echo 'Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst'
> Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> + curl -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> ++ type -P curl
> + /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> ++ cat /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> + [[ MD5= 73f41a7eacd77fe99fdf33c56ea4b8d0
> SHA1= e47119495c4e73658f13e817b0b552ac17f8cf5d
> SHA2-256= 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
> SHA2-512= 12dcb222b7e683d856f94b59ac1013943a55a413c2087b9ccdff639b420826edaadbd44c236f31a6f307a94adc3c3ae955d602219a6373019309c8012e6d2fd0 == \N\o\t\ \F\o\u\n\d ]]
> + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> ++ sed 's/ .*//'
> ++ md5sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> + SUM=73f41a7eacd77fe99fdf33c56ea4b8d0
> ++ uniq
> ++ grep 73f41a7eacd77fe99fdf33c56ea4b8d0 -o -a
> ++ grep MD5 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> + CHECKSUM=73f41a7eacd77fe99fdf33c56ea4b8d0
> + [[ 73f41a7eacd77fe99fdf33c56ea4b8d0 != \7\3\f\4\1\a\7\e\a\c\d\7\7\f\e\9\9\f\d\f\3\3\c\5\6\e\a\4\b\8\d\0 ]]
> + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> ++ sed 's/ .*//'
> ++ sha1sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> + SUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
> ++ uniq
> ++ grep e47119495c4e73658f13e817b0b552ac17f8cf5d -o -a
> ++ grep SHA1 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> + CHECKSUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
> + [[ e47119495c4e73658f13e817b0b552ac17f8cf5d != \e\4\7\1\1\9\4\9\5\c\4\e\7\3\6\5\8\f\1\3\e\8\1\7\b\0\b\5\5\2\a\c\1\7\f\8\c\f\5\d ]]
> + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> ++ sed 's/ .*//'
> ++ sha256sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> + SUM=56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
> ++ uniq
> ++ grep 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a -o -a
> ++ grep SHA256 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> + CHECKSUM=
> + [[ 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a != '' ]]
> + echo 'error: Check failed! Please check your network or try again.'
> error: Check failed! Please check your network or try again.
> + return 1
> + [[ 1 -eq 1 ]]
> + rm -r /tmp/tmp.2PjHux83CJ
> + echo 'removed: /tmp/tmp.2PjHux83CJ'
> removed: /tmp/tmp.2PjHux83CJ
> + exit 1
> ```

翻了下原代码，我觉得没必要把每种检验和都查一遍……你可以先手动把代码里下面这部分注释掉。
https://github.com/v2fly/fhs-install-v2ray/blob/a9c7b8d4912b1fd184fdac908d3884972d8b0df7/install-release.sh#L304-L312

**Comment by flben233** [comment]
> > > `bash -x ./install-release.sh --version v5.2.1`
> > > 试试呢
> > 
> > 
> > 结果是这样的
> > ```
> > root@shirakawatyu:~# bash -x ./install-release.sh --version v5.2.1
> > + DAT_PATH=/usr/local/share/v2ray
> > + JSON_PATH=/usr/local/etc/v2ray
> > + main --version v5.2.1
> > + check_if_running_as_root
> > + [[ 0 -ne 0 ]]
> > + identify_the_operating_system_and_architecture
> > ++ uname
> > + [[ Linux == \L\i\n\u\x ]]
> > + case "$(uname -m)" in
> > ++ uname -m
> > + MACHINE=64
> > + [[ ! -f /etc/os-release ]]
> > + [[ -f /.dockerenv ]]
> > + grep -q 'docker\|lxc' /proc/1/cgroup
> > + [[ -d /run/systemd/system ]]
> > + true
> > ++ type -P apt
> > + [[ -n /usr/bin/apt ]]
> > + PACKAGE_MANAGEMENT_INSTALL='apt -y --no-install-recommends install'
> > + PACKAGE_MANAGEMENT_REMOVE='apt purge'
> > + package_provide_tput=ncurses-bin
> > + judgment_parameters --version v5.2.1
> > + [[ 2 -gt 0 ]]
> > + case "$1" in
> > + VERSION=v5.2.1
> > + break
> > + install_software ncurses-bin tput
> > + package_name=ncurses-bin
> > + file_to_detect=tput
> > + type -P tput
> > + return
> > ++ tput setaf 1
> > + red=''
> > ++ tput setaf 2
> > + green=''
> > ++ tput setaf 6
> > + aoi=''
> > ++ tput sgr0
> > + reset=''
> > + [[ '' -eq 1 ]]
> > + [[ '' -eq 1 ]]
> > + [[ '' -eq 1 ]]
> > ++ mktemp -d
> > + TMP_DIRECTORY=/tmp/tmp.2PjHux83CJ
> > + ZIP_FILE=/tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> > + [[ '' -eq 1 ]]
> > + install_software curl curl
> > + package_name=curl
> > + file_to_detect=curl
> > + type -P curl
> > + return
> > + get_version
> > + [[ -n v5.2.1 ]]
> > + RELEASE_VERSION=v5.2.1
> > + return 2
> > + NUMBER=2
> > + [[ 2 -eq 0 ]]
> > + [[ '' -eq 1 ]]
> > + [[ 2 -eq 2 ]]
> > ++ uname -m
> > + echo 'info: Installing V2Ray v5.2.1 for x86_64'
> > info: Installing V2Ray v5.2.1 for x86_64
> > + download_v2ray
> > + DOWNLOAD_LINK=https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> > + echo 'Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip'
> > Downloading V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> > + curl -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> > ++ type -P curl
> > + /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -R -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip
> >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> >                                  Dload  Upload   Total   Spent    Left  Speed
> >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> > 100 11.0M  100 11.0M    0     0  15.8M      0 --:--:-- --:--:-- --:--:-- 57.1M
> > + echo 'Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst'
> > Downloading verification file for V2Ray archive: https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> > + curl -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> > ++ type -P curl
> > + /usr/bin/curl -L -q --retry 5 --retry-delay 10 --retry-max-time 60 -x '' -sSR -H 'Cache-Control: no-cache' -o /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst https://github.com/v2fly/v2ray-core/releases/download/v5.2.1/v2ray-linux-64.zip.dgst
> > ++ cat /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> > + [[ MD5= 73f41a7eacd77fe99fdf33c56ea4b8d0
> > SHA1= e47119495c4e73658f13e817b0b552ac17f8cf5d
> > SHA2-256= 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
> > SHA2-512= 12dcb222b7e683d856f94b59ac1013943a55a413c2087b9ccdff639b420826edaadbd44c236f31a6f307a94adc3c3ae955d602219a6373019309c8012e6d2fd0 == \N\o\t\ \F\o\u\n\d ]]
> > + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> > ++ sed 's/ .*//'
> > ++ md5sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> > + SUM=73f41a7eacd77fe99fdf33c56ea4b8d0
> > ++ uniq
> > ++ grep 73f41a7eacd77fe99fdf33c56ea4b8d0 -o -a
> > ++ grep MD5 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> > + CHECKSUM=73f41a7eacd77fe99fdf33c56ea4b8d0
> > + [[ 73f41a7eacd77fe99fdf33c56ea4b8d0 != \7\3\f\4\1\a\7\e\a\c\d\7\7\f\e\9\9\f\d\f\3\3\c\5\6\e\a\4\b\8\d\0 ]]
> > + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> > ++ sed 's/ .*//'
> > ++ sha1sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> > + SUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
> > ++ uniq
> > ++ grep e47119495c4e73658f13e817b0b552ac17f8cf5d -o -a
> > ++ grep SHA1 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> > + CHECKSUM=e47119495c4e73658f13e817b0b552ac17f8cf5d
> > + [[ e47119495c4e73658f13e817b0b552ac17f8cf5d != \e\4\7\1\1\9\4\9\5\c\4\e\7\3\6\5\8\f\1\3\e\8\1\7\b\0\b\5\5\2\a\c\1\7\f\8\c\f\5\d ]]
> > + for LISTSUM in 'md5' 'sha1' 'sha256' 'sha512'
> > ++ sed 's/ .*//'
> > ++ sha256sum /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip
> > + SUM=56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a
> > ++ uniq
> > ++ grep 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a -o -a
> > ++ grep SHA256 /tmp/tmp.2PjHux83CJ/v2ray-linux-64.zip.dgst
> > + CHECKSUM=
> > + [[ 56eb8d4727b058d10f8ff830bb0121381386b0695171767f38ba410f2613fc9a != '' ]]
> > + echo 'error: Check failed! Please check your network or try again.'
> > error: Check failed! Please check your network or try again.
> > + return 1
> > + [[ 1 -eq 1 ]]
> > + rm -r /tmp/tmp.2PjHux83CJ
> > + echo 'removed: /tmp/tmp.2PjHux83CJ'
> > removed: /tmp/tmp.2PjHux83CJ
> > + exit 1
> > ```
> 
> 翻了下原代码，我觉得没必要把每种检验和都查一遍……你可以先手动把代码里下面这部分注释掉。
> 
> https://github.com/v2fly/fhs-install-v2ray/blob/a9c7b8d4912b1fd184fdac908d3884972d8b0df7/install-release.sh#L304-L312

注释掉之后确实可以正常安装了

**Comment by qonmnop** [comment]
dgst文件的key发生了变化。

**Comment by LANPAST** [comment]
我的版本是5.1.0,更新到5.2.1的时候也遇到了这个问题

**Comment by yuehanwali** [comment]
不能把代码更正一下再发布出来吗

**Comment by Childdreams** [comment]
v2ray-linux-64.zip.dgst 里面没有SHA256 字符  有的是SHA2-256

**Comment by IceCodeNew** [comment]
fixed in #263
